spring-boot-project/spring-boot/src/main/java/org/springframework/boot/SpringApplication.java

```diff
@@ -1367,24 +1366,22 @@ public static int exit(ApplicationContext context, ExitCodeGenerator... exitCode
 		Assert.notNull(context, "Context must not be null");
 		int exitCode = 0;
 		try {
-			try {
-				ExitCodeGenerators generators = new ExitCodeGenerators();
-				Collection<ExitCodeGenerator> beans = context.getBeansOfType(ExitCodeGenerator.class).values();
-				generators.addAll(exitCodeGenerators);
-				generators.addAll(beans);
-				exitCode = generators.getExitCode();
-				if (exitCode != 0) {
-					context.publishEvent(new ExitCodeEvent(context, exitCode));
-				}
-			}
-			finally {
-				close(context);
+			ExitCodeGenerators generators = new ExitCodeGenerators();
+			Collection<ExitCodeGenerator> beans = context.getBeansOfType(ExitCodeGenerator.class).values();
+			generators.addAll(exitCodeGenerators);
+			generators.addAll(beans);
+			exitCode = generators.getExitCode();
+			if (exitCode != 0) {
+				context.publishEvent(new ExitCodeEvent(context, exitCode));
 			}
 		}
 		catch (Exception ex) {
 			ex.printStackTrace();
 			exitCode = (exitCode != 0) ? exitCode : 1;
 		}
+		finally {
+			close(context);
```

Comment:
This also changes behaviour as any exception thrown from `close(context)` will no longer be caught.